tests/Application/Actions/Hooks/StripeChargeUpdateTest.php

```diff
@@ -367,6 +428,73 @@ public function testSuccessfulFullRefund(): void
         $webhookSecret = $container->get('settings')['stripe']['accountWebhookSecret'];
         $time = (string) time();
 
+        $chatterProphecy = $this->prophesize(StripeChatterInterface::class);
+
+        // Double-check that the normal success case isn't messaging Slack.
+        $chatterProphecy->send(Argument::type(ChatMessage::class))->shouldNotBeCalled();
+
+        $donationRepoProphecy = $this->prophesize(DonationRepository::class);
+
+        $donationRepoProphecy
+            ->findOneBy(['chargeId' => 'ch_externalId_123'])
+            ->willReturn($donation)
+            ->shouldBeCalledOnce();
+
+        $donationRepoProphecy
+            ->releaseMatchFunds($donation)
+            ->shouldBeCalledOnce();
+
+        $donationRepoProphecy
+            ->push(Argument::type(Donation::class), false)
+            ->willReturn(true)
+            ->shouldBeCalledOnce();
+
+        $entityManagerProphecy = $this->prophesize(EntityManagerInterface::class);
+
+        $container->set(EntityManagerInterface::class, $entityManagerProphecy->reveal());
+        $container->set(DonationRepository::class, $donationRepoProphecy->reveal());
+        $container->set(StripeChatterInterface::class, $chatterProphecy->reveal());
+
+        // act
+        $request = $this->createRequest('POST', '/hooks/stripe', $body)
+            ->withHeader('Stripe-Signature', $this->generateSignature($time, $body, $webhookSecret));
+
+        $response = $app->handle($request);
+
+        // assert
+        $this->assertEquals('ch_externalId_123', $donation->getChargeId());
+        $this->assertEquals(DonationStatus::Refunded, $donation->getDonationStatus());
+        $this->assertEquals('1.00', $donation->getTipAmount());
+        $this->assertEquals(200, $response->getStatusCode());
+    }
+
+    /**
+     * Verifies that:
+     * * Slack gets a warning with the correct details
+     * * The donation record is updated with the new status, and
+     * * Match funds are released.
+     */
+    public function testOverRefundSendsSlackNoticeAndUpdatesRecord(): void
+    {
+        // arrange
+        $app = $this->getAppInstance();
+        /** @var Container $container */
+        $container = $app->getContainer();
```

Comment:
Probably not worth changing now, but I wonder if using the container and the app instance actually gives us much in a unit test? I feel like its sometimes easier to see exactly what the code being tested is if the test directly instantiates e.g. the `StripeChargeUpdate` class and passes in the dependencies.  Or if we want to test the app as a whole maybe go the other way and do an integration test with the real DB then we test the repository code too.